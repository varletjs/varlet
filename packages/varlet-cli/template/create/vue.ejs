<template>
  <div class="var-<%- kebabCaseName %>"></div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import './<%- kebabCaseName %>.less'

export default defineComponent({
  name: '<%- namespace + bigCamelizeName %>',
})
</script>
