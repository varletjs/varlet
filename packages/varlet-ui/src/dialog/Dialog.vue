<template>
  <var-popup
    class="var-dialog__popup"
    var-dialog-cover
    :show="popupShow"
    :overlay="overlay"
    :overlay-class="overlayClass"
    :overlay-style="overlayStyle"
    :lock-scroll="lockScroll"
    :close-on-click-overlay="popupCloseOnClickOverlay"
    :teleport="teleport"
    @open="onOpen"
    @close="onClose"
    @closed="onClosed"
    @opened="onOpened"
    @route-change="onRouteChange"
    @click-overlay="handleClickOverlay"
  >
    <div class="var--box var-dialog" :class="dialogClass" :style="dialogStyle" v-bind="$attrs">
      <div class="var-dialog__title">
        <slot name="title">{{ dt(title, pack.dialogTitle) }}</slot>
      </div>
      <div class="var-dialog__message" :style="{ textAlign: messageAlign }">
        <slot>
          {{ message }}
        </slot>
      </div>
      <div class="var-dialog__actions">
        <var-button
          class="var-dialog__button var-dialog__cancel-button"
          var-dialog-cover
          text
          :text-color="cancelButtonTextColor"
          :color="cancelButtonColor"
          v-if="cancelButton"
          @click="cancel"
        >
          {{ dt(cancelButtonText, pack.dialogCancelButtonText) }}
        </var-button>
        <var-button
          class="var-dialog__button var-dialog__confirm-button"
          var-dialog-cover
          text
          :text-color="confirmButtonTextColor"
          :color="confirmButtonColor"
          v-if="confirmButton"
          @click="confirm"
        >
          {{ dt(confirmButtonText, pack.dialogConfirmButtonText) }}
        </var-button>
      </div>
    </div>
  </var-popup>
</template>

<script lang="ts">
import VarPopup from '../popup'
import VarButton from '../button'
import { props } from './props'
import { defineComponent, ref, watch } from 'vue'
import { dt } from '../utils/shared'
import { pack } from '../locale'
import type { Ref } from 'vue'

export default defineComponent({
  name: 'VarDialog',
  components: {
    VarPopup,
    VarButton,
  },
  inheritAttrs: false,
  props,
  setup(props) {
    const popupShow: Ref<boolean> = ref(false)
    const popupCloseOnClickOverlay: Ref<boolean> = ref(false)

    const done = () => props['onUpdate:show']?.(false)

    const handleClickOverlay = () => {
      const { closeOnClickOverlay, onClickOverlay, onBeforeClose } = props

      onClickOverlay?.()

      if (!closeOnClickOverlay) {
        return
      }

      if (onBeforeClose != null) {
        onBeforeClose('close', done)
        return
      }

      props['onUpdate:show']?.(false)
    }

    const confirm = () => {
      const { onBeforeClose, onConfirm } = props

      onConfirm?.()

      if (onBeforeClose != null) {
        onBeforeClose('confirm', done)
        return
      }

      props['onUpdate:show']?.(false)
    }

    const cancel = () => {
      const { onBeforeClose, onCancel } = props

      onCancel?.()

      if (onBeforeClose != null) {
        onBeforeClose('cancel', done)
        return
      }

      props['onUpdate:show']?.(false)
    }

    watch(
      () => props.show,
      (newValue) => {
        popupShow.value = newValue
      },
      { immediate: true }
    )

    watch(
      () => props.closeOnClickOverlay,
      (newValue) => {
        if (props.onBeforeClose != null) {
          popupCloseOnClickOverlay.value = false
          return
        }

        popupCloseOnClickOverlay.value = newValue
      },
      { immediate: true }
    )

    return {
      pack,
      dt,
      popupShow,
      popupCloseOnClickOverlay,
      handleClickOverlay,
      confirm,
      cancel,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../popup/popup';
@import '../button/button';
@import './dialog';
</style>
