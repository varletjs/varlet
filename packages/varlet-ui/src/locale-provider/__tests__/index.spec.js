import { createApp, h } from 'vue'
import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import DatePicker from '../../date-picker'
import Dialog from '../../dialog/Dialog.vue'
import Locale from '../../locale'
import LocaleProvider from '../index'

test('localeProvider component plugin', () => {
  const app = createApp({}).use(LocaleProvider)
  expect(app.component(LocaleProvider.name)).toBeTruthy()
})

test('localeProvider tag', async () => {
  const wrapper = mount(LocaleProvider)
  expect(wrapper.html()).toMatchSnapshot()
  await wrapper.setProps({ tag: 'span' })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('localeProvider locale', async () => {
  const container = document.createElement('div')
  const wrapper = mount(LocaleProvider, {
    props: {
      messages: {
        'zh-CN': Locale.zhCN,
        'en-US': Locale.enUS,
      },
    },
    slots: {
      default: () =>
        h('div', [h(Dialog, { show: true, teleport: container }), h(DatePicker, { modelValue: '2021-04-08' })]),
    },
  })
  expect(container.innerHTML).toMatchSnapshot()
  expect(wrapper.html()).toMatchSnapshot()
  expect(container.innerHTML).contains('确认')
  expect(container.innerHTML).contains('取消')
  expect(container.innerHTML).contains('提示')
  await wrapper.setProps({ locale: 'en-US' })
  expect(container.innerHTML).toMatchSnapshot()
  expect(wrapper.html()).toMatchSnapshot()
  expect(container.innerHTML).contains('Confirm')
  expect(container.innerHTML).contains('Cancel')
  expect(container.innerHTML).contains('Hint')

  wrapper.unmount()
})
