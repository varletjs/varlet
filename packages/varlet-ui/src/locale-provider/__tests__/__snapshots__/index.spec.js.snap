// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`localeProvider locale 1`] = `"<transition-stub name="var-fade" appear="false" persisted="false" css="true"><div class="var--box var-popup" style="z-index: 1998;"><div class="var-popup__overlay" style="z-index: 1999;"></div><transition-stub name="var-pop-center" appear="false" persisted="false" css="true"><div class="var-popup__content var-popup--center var-popup--content-background-color var-elevation--3 var-dialog__popup" style="z-index: 2000;" role="dialog" aria-modal="true" var-dialog-cover=""><div class="var--box var-dialog"><div class="var-dialog__title">提示</div><div class="var-dialog__message" style="text-align: left;"></div><div class="var-dialog__actions"><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-dialog__button var-dialog__cancel-button" type="button" var-dialog-cover=""><!--v-if--><div class="var-button__content">取消</div><div class="var-hover-overlay"></div></button><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-dialog__button var-dialog__confirm-button" type="button" var-dialog-cover=""><!--v-if--><div class="var-button__content">确认</div><div class="var-hover-overlay"></div></button></div></div></div></transition-stub></div></transition-stub>"`;

exports[`localeProvider locale 2`] = `
"<div class="var-locale-provider">
  <div>
    <!--teleport start-->
    <!--teleport end-->
    <div class="var-date-picker">
      <div class="var-date-picker__title">
        <div class="var-date-picker__title-select">
          <div class="var-date-picker__title-hint">选择日期</div>
          <div class="var-date-picker__title-year">2021</div>
        </div>
        <div class="var-date-picker__title-date var-date-picker__title-date--active">
          <transition-stub name="var-date-picker-translatey" appear="false" persisted="false" css="true">
            <div>04-08 星期四</div>
          </transition-stub>
        </div>
      </div>
      <div class="var-date-picker__body">
        <transition-stub name="var-date-picker-panel-fade" appear="false" persisted="false" css="true">
          <div class="var-day-picker">
            <div class="var-day-picker__content">
              <div class="var-date-picker-header"><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-button--round var-date-picker-header__arrow" type="button" var-date-picker-header-cover="">
                  <!--v-if-->
                  <div class="var-button__content"><i class="var-icon var-icon--set var-icon-chevron-left" style="transition-duration: 0ms;"></i></div>
                  <div class="var-hover-overlay"></div>
                </button>
                <div class="var-date-picker-header__value">
                  <transition-stub name="var-date-picker-translatex" appear="false" persisted="false" css="true">
                    <div>2021 四月</div>
                  </transition-stub>
                </div><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-button--round var-date-picker-header__arrow" type="button" var-date-picker-header-cover="">
                  <!--v-if-->
                  <div class="var-button__content"><i class="var-icon var-icon--set var-icon-chevron-right" style="transition-duration: 0ms;"></i></div>
                  <div class="var-hover-overlay"></div>
                </button>
              </div>
              <transition-stub name="var-date-picker-translatex" appear="false" persisted="false" css="true">
                <div>
                  <ul class="var-day-picker__head">
                    <li>日</li>
                    <li>一</li>
                    <li>二</li>
                    <li>三</li>
                    <li>四</li>
                    <li>五</li>
                    <li>六</li>
                  </ul>
                  <ul class="var-day-picker__body"></ul>
                </div>
              </transition-stub>
            </div>
          </div>
        </transition-stub>
      </div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`localeProvider locale 3`] = `"<transition-stub name="var-fade" appear="false" persisted="false" css="true"><div class="var--box var-popup" style="z-index: 1998;"><div class="var-popup__overlay" style="z-index: 1999;"></div><transition-stub name="var-pop-center" appear="false" persisted="false" css="true"><div class="var-popup__content var-popup--center var-popup--content-background-color var-elevation--3 var-dialog__popup" style="z-index: 2000;" role="dialog" aria-modal="true" var-dialog-cover=""><div class="var--box var-dialog"><div class="var-dialog__title">Hint</div><div class="var-dialog__message" style="text-align: left;"></div><div class="var-dialog__actions"><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-dialog__button var-dialog__cancel-button" type="button" var-dialog-cover=""><!--v-if--><div class="var-button__content">Cancel</div><div class="var-hover-overlay"></div></button><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-dialog__button var-dialog__confirm-button" type="button" var-dialog-cover=""><!--v-if--><div class="var-button__content">Confirm</div><div class="var-hover-overlay"></div></button></div></div></div></transition-stub></div></transition-stub>"`;

exports[`localeProvider locale 4`] = `
"<div class="var-locale-provider">
  <div>
    <!--teleport start-->
    <!--teleport end-->
    <div class="var-date-picker">
      <div class="var-date-picker__title">
        <div class="var-date-picker__title-select">
          <div class="var-date-picker__title-hint">SELECT DATE</div>
          <div class="var-date-picker__title-year">2021</div>
        </div>
        <div class="var-date-picker__title-date var-date-picker__title-date--active">
          <transition-stub name="var-date-picker-translatey" appear="false" persisted="false" css="true">
            <div>Thu, Apr 8</div>
          </transition-stub>
        </div>
      </div>
      <div class="var-date-picker__body">
        <transition-stub name="var-date-picker-panel-fade" appear="false" persisted="false" css="true">
          <div class="var-day-picker">
            <div class="var-day-picker__content">
              <div class="var-date-picker-header"><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-button--round var-date-picker-header__arrow" type="button" var-date-picker-header-cover="">
                  <!--v-if-->
                  <div class="var-button__content"><i class="var-icon var-icon--set var-icon-chevron-left" style="transition-duration: 0ms;"></i></div>
                  <div class="var-hover-overlay"></div>
                </button>
                <div class="var-date-picker-header__value">
                  <transition-stub name="var-date-picker-translatex" appear="false" persisted="false" css="true">
                    <div>April 2021</div>
                  </transition-stub>
                </div><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-default var-button--round var-date-picker-header__arrow" type="button" var-date-picker-header-cover="">
                  <!--v-if-->
                  <div class="var-button__content"><i class="var-icon var-icon--set var-icon-chevron-right" style="transition-duration: 0ms;"></i></div>
                  <div class="var-hover-overlay"></div>
                </button>
              </div>
              <transition-stub name="var-date-picker-translatex" appear="false" persisted="false" css="true">
                <div>
                  <ul class="var-day-picker__head">
                    <li>S</li>
                    <li>M</li>
                    <li>T</li>
                    <li>W</li>
                    <li>T</li>
                    <li>F</li>
                    <li>S</li>
                  </ul>
                  <ul class="var-day-picker__body">
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-button--disabled var-button--text-disabled var-day-picker__button" type="button" disabled="" var-day-picker-cover="">
                        <!--v-if-->
                        <div class="var-button__content"></div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-button--disabled var-button--text-disabled var-day-picker__button" type="button" disabled="" var-day-picker-cover="">
                        <!--v-if-->
                        <div class="var-button__content"></div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-button--disabled var-button--text-disabled var-day-picker__button" type="button" disabled="" var-day-picker-cover="">
                        <!--v-if-->
                        <div class="var-button__content"></div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-button--disabled var-button--text-disabled var-day-picker__button" type="button" disabled="" var-day-picker-cover="">
                        <!--v-if-->
                        <div class="var-button__content"></div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">1</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">2</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">3</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">4</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">5</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">6</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">7</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-elevation--2 var-button--primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="false">
                        <!--v-if-->
                        <div class="var-button__content">8</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">9</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">10</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">11</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">12</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">13</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">14</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">15</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">16</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">17</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">18</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">19</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">20</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">21</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">22</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">23</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">24</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">25</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">26</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">27</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">28</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">29</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                    <li><button class="var-button var--box var-button--normal var--inline-flex var-button--text var-button--text-primary var-button--round var-day-picker__button var-day-picker__button--usable" type="button" var-day-picker-cover="" var-date-picker-color-cover="true">
                        <!--v-if-->
                        <div class="var-button__content">30</div>
                        <div class="var-hover-overlay"></div>
                      </button></li>
                  </ul>
                </div>
              </transition-stub>
            </div>
          </div>
        </transition-stub>
      </div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`localeProvider tag 1`] = `"<div class="var-locale-provider"></div>"`;

exports[`localeProvider tag 2`] = `"<span class="var-locale-provider"></span>"`;
